src/mongocxx/config/compiler.hpp

```diff
@@ -0,0 +1,15 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define MONGOCXX_INLINE inline __attribute__((__visibility__("hidden"), __always_inline__))
```

Comment:
We could do something like:
```
#include <bsoncxx/config/prelude.hpp>

#define MONGOCXX_INLINE BSONCXX_INLINE

#include <bsoncxx/config/postlude.hpp>
```

Do you think that's a better solution? I had thought about this but it seemed a bit overkill, though more readable.